rosbag2_compression/src/rosbag2_compression/zstd_compressor.cpp

```diff
@@ -34,6 +34,8 @@ constexpr const int kDefaultZstdCompressionLevel = 1;
 
 // String constant used to identify ZstdCompressor.
 constexpr const char kCompressionIdentifier[] = "zstd";
+// Used as a parameter type in a function that accepts the output of ZSTD_compress.
+using ZstdCompressReturnType = decltype(ZSTD_compress(nullptr, 0, nullptr, 0, 0));
```

Comment:
Maybe use https://en.cppreference.com/w/cpp/types/result_of instead of making a function call.

Reply:
decltype actually doesn't make a function call! (I'll update with the document stating this)

Reply:
Never mind. `decltype` doesn't invoke anything, but `result_of` is better in this situation because you won't need to write the dummy values to "pass" into the `ZSTD_compress()` function.

Reply:
Yeah, I agree, unfortunately I merged this before you commented.